master/internal/rm/multirm/multirm_intg_test.go

```diff
@@ -36,6 +39,8 @@ func TestMain(m *testing.M) {
 		},
 		syslog: logrus.WithField("component", "resource-router"),
 	}
+
+	os.Exit(m.Run())
```

Comment:
I think this is a weird usage of ```TestMain``` and global variables 

I think each function should just call a helper to create this mock resource manager

Reply:
on the contrary, I think defining global variables used by every single test should belong in TestMain

Reply:
why even define the global variable? It saves like one line of setup code but introduces all this dependency between different tests

This is kinda separate but I still really don't think TestMain adds any value for the complexity it can add